compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -788,8 +788,13 @@ impl<'context> Elaborator<'context> {
             | Type::Bool => (),
 
             Type::TypeVariable(_, _) => {
-                self.push_err(TypeCheckError::TypeAnnotationsNeeded { span });
-                return Type::Error;
+                // NOTE: in reality the expected type can also include bool, but for the compiler's simplicity
+                // we only allow integer types. If a bool is in `from` it will need an explicit type annotation.
+                let expected = &&Type::polymorphic_integer_or_field(self.interner);
+                self.unify(from, expected, || TypeCheckError::InvalidCast {
```

Comment:
```suggestion
                let expected = Type::polymorphic_integer_or_field(self.interner);
                self.unify(from, &expected, || TypeCheckError::InvalidCast {
```
One two many references